doc/task-lists-and-tasks.md

```diff
@@ -0,0 +1,185 @@
+
+```
+[
+    {
+        identifier: :project_kick_off,
+        tasks: [
+                Conversion::Voluntary::Tasks::Handover,
+                Conversion::Voluntary::Tasks::StakeholderKickOff,
+                Conversion::Voluntary::Tasks::ConversionGrant
+                ]
+    },
+]
+```
+
+You'll see how each task list groups its own tasks in a `tasks` directory.
+
+With the new task list in place, you must declare it in the `delegate_type` on
+the Project model, which will force each project to have a `TaskList`
+association, returning the correct type of task list.
+
+## Tasks
+
+### Adding a new task
+
+With a task list in place, developers can populate them with tasks.
+
+Each task inherits from the `TaskList::Task` or `Tasklist::OptionalTask` classes
+and declares the attributes that make up the actions in the task.
+
+Conventionally the Task name reflects the actual task the user is performing:
+
+`Conversion::Voluntary::Tasks::CheckBaseline` : Check the Baseline
+`Conversion::Voluntary::Tasks::OneHundredAndTwentyFiveYearLease` : 125 year
+lease
+
+The actual words to describe the task and any content for a task list or task
+are loaded from the locales, convention lets Rails load the correct content
+based on the class names used.
+
+The actions for each task are declared as attributes of the task but are stored
+in the task lists table.
+
+Conventionally, each action must start with the task class name followed by a
+meaningful identifier, much like tasks themselves:
+
+Task: `Conversion::Voluntary::Tasks::Handover`
+
+Task attributes:
+
+```
+attribute :review
+attribute :notes
+attribute :meeting
+```
+
+Database columns:
+
+```
+conversion_voluntary_tasks_review
+conversion_voluntary_tasks_notes
+conversion_voluntary_tasks_meeting
+```
+
+Once again, following the naming convention used allows all of the content to be
+loaded from the locales.
+
+Each new task requires a number of files to be created:
+
+- the model
+- the database migration
+- the view
+- the locales
+
+#### Rails generators
+
+To make this task simpler Rails generators have been created for the current
+task lists:
+
+Run the generator and supply the task class name:
+
+```
+bin/rails generate conversion_voluntary_task NameOfTask
+```
+
+Right now there is a generator per task list, the example above would create:
+
+`Conversion::Voluntary::Tasks::NameOfTask` and the relevant files:
+
+```
+app/models/conversion/voluntary/tasks/name_of_task.rb
+app/views/conversions/voluntary/task_lists/tasks/name_of_task.html.rb
+config/locales/task_lists/conversion/voluntary/name_of_task.en.yml
+```
+
+And the same for involuntary conversions:
+
+```
+bin/rails generate conversion_involuntary_task NameOfTask
+```
+
+Each task and action has a predefined set of locale keys which will be used
+based on the naming conventions, see existing task locale files for more
+details.
+
+### A note on notes
+
+As there is no database representation of a task, the task identifier, it's
```

Comment:
Grammar suggestion: `the task identifier - its class name - is used as a unique identifier`

(`It's` is only ever a contraction of `it is`; and the hyphens instead of commas make this sentence easier to read)